core/src/Lender.sol

```diff
@@ -137,7 +137,13 @@ contract Lender is Ledger {
         emit Deposit(msg.sender, beneficiary, amount, shares);
     }
 
-    function redeem(uint256 shares, address recipient, address owner) external returns (uint256 amount) {
+    // TODO: We aren't `require`ing that the return value of `deposit` == `shares`. Check it in fuzz tests.
```

Comment:
Could you expand on this?

Reply:
In theory, we could add an additional `require` that the return value of deposit is equal to the original passed-in `shares`. But as long as the `previewMint` function is up to spec, that's guaranteed anyway. So instead of adding the `require` and wasting gas, I added the TODO for us to verify `previewMint` functionality with fuzz testing. (We actually already test `previewDeposit` and `previewRedeem` so it won't be too much work to add `previewMint` and `previewWithdraw`)